escheduler-server/src/main/java/cn/escheduler/server/worker/task/http/HttpTask.java

```diff
@@ -0,0 +1,310 @@
+            long costTime = System.currentTimeMillis() - startTime;
+            status = StringUtils.isEmpty(output);
+            logger.info("startTime: {}, httpUrl: {}, httpMethod: {}, status : {}, costTime : {}Millisecond, statusCode : {}, body : {}, log : {}",
+                    DateUtils.format2Readable(startTime), httpParameters.getUrl(),httpParameters.getHttpMethod(), status, costTime, statusCode, body, output);
+        } catch (Exception e) {
+            appendMessage(e.toString());
+            exitStatusCode = -1;
+            logger.error("httpUrl[" + httpParameters.getUrl() + "] connection failed："+output, e);
+        }
+    }
+
+    protected CloseableHttpResponse sendRequest() throws ClientProtocolException, IOException {
+        RequestBuilder builder = createRequestBuilder();
+        ProcessInstance processInstance = processDao.findProcessInstanceByTaskId(taskProps.getTaskInstId());
+
+        Map<String, Property> paramsMap = ParamUtils.convert(taskProps.getUserDefParamsMap(),
+                taskProps.getDefinedParams(),
+                httpParameters.getLocalParametersMap(),
+                processInstance.getCmdTypeIfComplement(),
+                processInstance.getScheduleTime());
+
+        addRequestParams(builder,paramsMap);
+        HttpUriRequest request = builder.setUri(httpParameters.getUrl()).build();
+        setHeaders(request,paramsMap);
+        CloseableHttpClient client = createHttpClient();
+        return client.execute(request);
+    }
+
+    protected String getResponseBody(CloseableHttpResponse httpResponse) throws ParseException, IOException {
+        if (httpResponse == null) {
+            return null;
+        }
+        HttpEntity entity = httpResponse.getEntity();
+        if (entity == null) {
+            return null;
+        }
+        String webPage = EntityUtils.toString(entity, "UTF-8");
+        return webPage;
+    }
+
+    protected int getStatusCode(CloseableHttpResponse httpResponse) {
+        int status = httpResponse.getStatusLine().getStatusCode();
+        return status;
+    }
+
+    protected int validResponse(String body, String statusCode) throws Exception {
+        int exitStatusCode = 0;
+        switch (httpParameters.getHttpCheckCondition()) {
+            case BODY_CONTAINS:
+                if (StringUtils.isEmpty(body) || !body.contains(httpParameters.getCondition())) {
+                    appendMessage(httpParameters.getUrl() + " doesn contain "
+                            + XmlEscapers.xmlContentEscaper().escape(httpParameters.getCondition()));
+                    exitStatusCode = -1;
+                }
+                break;
+            case BODY_NOT_CONTAINS:
+                if (StringUtils.isEmpty(body) || body.contains(httpParameters.getCondition())) {
+                    appendMessage(httpParameters.getUrl() + " contains "
+                            + XmlEscapers.xmlContentEscaper().escape(httpParameters.getCondition()));
```

Comment:
Seems this XmlEscapes is not necessary, can we just don't escape?  (since the error message will be saved into JSON format and passed back to javascript side. And our javascript side should never just display raw HTML, so there is no need to escape XML